client/asset/btc/btc.go

```diff
@@ -1923,6 +1940,605 @@ func (btc *baseWallet) fundedTx(coins asset.Coins) (*wire.MsgTx, uint64, []outPo
+		}
+	}
+
+	baseTx, totalIn, _, err := btc.fundedTx(append(additionalInputs, orderChange))
+	if err != nil {
+		return makeError(err)
+	}
+
+	changeAddr, err := btc.node.changeAddress()
+	if err != nil {
+		return makeError(fmt.Errorf("error creating change address: %w", err))
+	}
+
+	tx, output, txFee, err := btc.signTxAndAddChange(baseTx, changeAddr, totalIn, additionalFeesRequired, newFeeRate)
+	if err != nil {
+		return makeError(err)
+	}
+
+	return tx, output, txFee + additionalFeesRequired, nil
+}
+
+// AccelerateOrder uses the Child-Pays-For-Parent technique to accelerate a
+// chain of swap transactions and previous accelerations. It broadcasts a new
+// transaction with a fee high enough so that the average fee of all the
+// unconfirmed transactions in the chain and the new transaction will have
+// an average fee rate of newFeeRate. The changeCoin argument is the latest
+// chhange in the order. It must be the input in the acceleration transaction
+// in order for the order to be accelerated. requiredForRemainingSwaps is the
+// amount of funds required to complete the rest of the swaps in the order.
+// The change output of the acceleration transaction will have at least
+// this amount.
+//
+// The returned change coin may be nil, and should be checked before use.
+func (btc *ExchangeWalletAccelerator) AccelerateOrder(swapCoins, accelerationCoins []dex.Bytes, changeCoin dex.Bytes, requiredForRemainingSwaps, newFeeRate uint64) (asset.Coin, string, error) {
+	return accelerateOrder(btc.baseWallet, swapCoins, accelerationCoins, changeCoin, requiredForRemainingSwaps, newFeeRate)
+}
+
+// AccelerateOrder uses the Child-Pays-For-Parent technique to accelerate a
+// chain of swap transactions and previous accelerations. It broadcasts a new
+// transaction with a fee high enough so that the average fee of all the
+// unconfirmed transactions in the chain and the new transaction will have
+// an average fee rate of newFeeRate. The changeCoin argument is the latest
+// chhange in the order. It must be the input in the acceleration transaction
+// in order for the order to be accelerated. requiredForRemainingSwaps is the
+// amount of funds required to complete the rest of the swaps in the order.
+// The change output of the acceleration transaction will have at least
+// this amount.
+//
+// The returned change coin may be nil, and should be checked before use.
+func (btc *ExchangeWalletSPV) AccelerateOrder(swapCoins, accelerationCoins []dex.Bytes, changeCoin dex.Bytes, requiredForRemainingSwaps, newFeeRate uint64) (asset.Coin, string, error) {
+	return accelerateOrder(btc.baseWallet, swapCoins, accelerationCoins, changeCoin, requiredForRemainingSwaps, newFeeRate)
+}
+
+func accelerateOrder(btc *baseWallet, swapCoins, accelerationCoins []dex.Bytes, changeCoin dex.Bytes, requiredForRemainingSwaps, newFeeRate uint64) (asset.Coin, string, error) {
+	btc.fundingMtx.Lock()
+	defer btc.fundingMtx.Unlock()
```

Comment:
This is OK I guess, and it's roughly the same amount of stuff happening under lock in e.g. `FundOrder` (esp. with split enabled), but in general, we're doing a ton with this mutex locked.  The broadcast can take ~seconds, and any RPC is generally "long" in terms of lock contention.  And this all blocks swaps on completely unrelated orders.

I have no suggestions how to change this given I think we want both the fundingCoins map and the results of listlockunspent fixed until the acceleration is sent.  It's just notable and something to consider reworking in the future.